Persistable playlists

bergquist opened the pull request:
Lets have the discussion about the persistable playlists here. @utkarshcmu @nikita-graf

closes #1137

utkarshcmu committed:
Resurrected nikita-graf's work and added playlistType for future use

utkarshcmu committed:
Removed unused components

utkarshcmu committed:
Added play button

utkarshcmu committed:
Made playlist URL clickable

utkarshcmu committed:
Playlist can be run without saving too

utkarshcmu committed:
Merged with master, resolved conflicts

utkarshcmu committed:
Fixed gofmt tests

bergquist committed:
Merge pull request #3611 from utkarshcmu/recorded

Playlist enhancements

utkarshcmu committed:
Made Playlist UI user friendly

bergquist committed:
Merge pull request #3613 from utkarshcmu/recorded

Made Playlist UI user friendly

bergquist committed:
feat(playlist): make default timespan placeholder

bergquist committed:
feat(playlist): improve header

bergquist committed:
feat(playlist): add search result count

bergquist committed:
feat(playlist): minor ux changes for playlists

nikita-graf committed:
remove dashboard from playlist when its destroyed

bergquist committed:
Merge pull request #3634 from nikita-graf/dashboard-remove-playlist-sync

Playlist improvements

nikita-graf committed:
basic playlist dashboards sort support

bergquist committed:
Merge pull request #3649 from nikita-graf/playlist-sort

Playlist dashboards sort

utkarshcmu commented:
Looks good to me... except what @nikita-graf said about the sorting arrows in #3649. We need to fix that before merging this to master. Rest everything looks awesome :)

utkarshcmu committed:
Improvized dashboard sorting

utkarshcmu committed:
Better heading

utkarshcmu commented:
@bergquist - 2 more improvements : #3664 & #3665 , if they make sense then please merge them. :)

bergquist committed:
Merge pull request #3665 from utkarshcmu/recorded2

Better heading

bergquist committed:
Merge pull request #3664 from utkarshcmu/recorded

Improvized dashboard sorting UI

utkarshcmu commented:
So, whats left in this PR from getting merged to master?

bergquist commented:
@utkarshcmu @nikita-graf Seems like one of you have not signed the CLA. Please do so at https://cla-assistant.io/grafana/grafana?pullRequest=3655

bergquist committed:
style(playlist): rename files to match new format

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=3655) <br/>All committers of the pull request should sign our Contributor License Agreement in order to get your pull request merged.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: bergquist<br/>:x: utkarshcmu<br/>:x: nikita-graf

bergquist committed:
style(playlists): use backendsrv direct

There is no point resolving the playlists in the routes file.
Loading playlists is what the controller is suppose to do

bergquist committed:
fix(playlist): move backend code to ctrl

data loading should be done in the ctrl

bergquist committed:
fix(playlist): implement api according to new standard

bergquist committed:
fix(playlist): move dashboard uri cration to Backend

bergquist committed:
fix(playlist): fix broken build. unused vars

bergquist committed:
feat(playlist): changes to relation table

Also introduces an abstraction between playlist and dashboard.
This will make it possible to att search, and tag filtering to
playlists without any major refactoring

nikita-graf commented:
Just signed it.

utkarshcmu committed:
Update playlist.md

utkarshcmu commented:
Updated docs #3690

bergquist committed:
feat(playlist): refactor FE to support playlistitems

bergquist committed:
Merge pull request #3690 from utkarshcmu/recorded3

Update playlist.md

bergquist committed:
fix(playlist): fixes broken search filter

bergquist committed:
fix(playlist): remove start button from edit page

bergquist committed:
fix(playlist): new url for viewing playlist

bergquist committed:
Merge branch 'playlist_relation_table' into playlist

bergquist committed:
style(playlist): abstract DTO creation

bergquist committed:
tests(playlist): adds ctrl test for playlist edit

bergquist committed:
fix(playlist): remove only parameter to describe

bergquist committed:
style(playlist): renames variable

bergquist committed:
feat(playlist): move playlistSrv to playlist folder

bergquist committed:
feat(playlist): renames timespan to interval

torkelo commented on pkg/api/api.go:
```diff
@@ -169,6 +172,17 @@ func Register(r *macaron.Macaron) {
+		r.Group("/playlists", func() {
+			r.Get("/", wrap(SearchPlaylists))
+			r.Get("/:id", ValidateOrgPlaylist, wrap(GetPlaylist))
+			r.Get("/:id/playlistitems", ValidateOrgPlaylist, wrap(GetPlaylistItems))
```
är inte `/playlists/1232/items`  enklare, vad är skillanden mellan items och dashboards?

bergquist committed:
fix(playlist): remove invalid logging

bergquist committed:
feat(playlist): shortens urls

bergquist committed:
fix(playlist): fix for timespan -> interval
